app/assets/javascripts/discourse/app/components/modal/sidebar-section-form.js

```diff
@@ -309,6 +309,23 @@ export default class SidebarSectionForm extends Component {
   }
 
   update() {
+    this.model?.section?.public || this.transformedModel.public
+      ? this.#updateWithConfirm()
+      : this.#updateCall();
+  }
+
+  #updateWithConfirm() {
+    return this.dialog.yesNoConfirm({
+      message: this.transformedModel.public
```

Comment:
I see you don't check `this.model?.section?.public` here but you do in `update`. Maybe we just make a shortcut and use that everywhere?

```javascript
get isPublic() {
  return this.model?.section?.public|| this.transformedModel.public
}
```

Reply:
I think it is correct
` this.model?.section?.public` is checking the state of the section when it was opened
`this.transformedModel.public` is checking state of section after user interaction

In the first step, we check the form was public or is now public
```
    this.model?.section?.public || this.transformedModel.public
      ? this.#updateWithConfirm()
      : this.#updateCall();
```

And then when we are in `#updateWithConfirm()` we need to determine the message type. `this.transformedModel.public` means that the current new version is public. Else means that it was public but became private.

Let me define 2 helper functions like `wasPublic` and `isPublic` so that code will be more clear

Reply:
Ah okay cool, thanks for adding the helpers :+1: